website/blog/2025-10-08-react-native-0.82.md

```diff
@@ -0,0 +1,313 @@
+
+## DOM Node APIs
+
+Starting from React Native 0.82, native components will provide DOM-like nodes via refs.
+
+Before, native components provided React Native-specific objects with just a handful of methods like `measure` and `setNativeProps`. After this release, they will provide [nodes implementing a subset of the DOM API](https://reactnative.dev/docs/element-nodes) that allow traversing the UI tree, measuring layout, etc. as they do on Web, e.g.:
+
+```jsx
+function MyComponent(props) {
+  const ref = useRef();
+
+  useEffect(() => {
+    const element = ref.current;
+
+    // New methods
+    element.parentNode;
+    element.parentElement;
+    element.childNodes;
+    element.children;
+    const bounds = element.getBoundingClientRect();
+    const doc = element.ownerDocument;
+    const maybeElement = doc.getElementById('some-view');
+
+    // Legacy methods are still available
+    element.measure((x, y, width, height, pageX, pageY) => {
+      /* ... */
+    });
+  }, []);
+
+  return <View ref={ref} />;
+}
+```
+
+Additionally, this will expose access to leaf [text nodes](https://reactnative.dev/docs/text-nodes) (created by the `Text` component) and [document nodes](https://reactnative.dev/docs/document-nodes) representing React Native root nodes.
+
+This is a backwards compatible change, as the new nodes will continue implementing the legacy methods (like `measure`).
+
+For more information, please check our [documentation](https://reactnative.dev/docs/nodes).
+
+## Other changes
+
+### Web Performance APIs (Canary)
+
+React Native now implements a subset of the performance APIs available on Web:
+
+- [High Resolution Time](https://www.w3.org/TR/hr-time-3/): defines `performance.now()` and `performance.timeOrigin`.
+- [Performance Timeline](https://w3c.github.io/performance-timeline/): defines `PerformanceObserver` and methods to access performance entries in the performance object (`getEntries()`, `getEntriesByType()`, `getEntriesByName()`).
+- [User Timing](https://w3c.github.io/user-timing/): defines `performance.mark` and `performance.measure`.
+- [Event Timing API](https://w3c.github.io/event-timing/): defines `event` entry types reported to `PerformanceObserver`.
+- [Long Tasks API](https://w3c.github.io/longtasks/): defines `longtask` entry types reported to `PerformanceObserver`.
+
+They allow tracking different aspects of performance in your app at runtime (for telemetry) and they will be visible in the performance panel in React Native DevTools (available in a future version of React Native).
+
+They are currently **available only in the [canary release level](https://reactnative.dev/docs/next/releases/release-levels)**, and will be released as stable in a future version of React Native.
```

Comment:
```suggestion
They are currently **available only in the [canary release level](/docs/next/releases/release-levels)**, and will be released as stable in a future version of React Native.
```

Internal link.

Reply:
are you sure I wanted all these to be opened in a new tab

Reply:
We can leave the full links if this is intentional :) 

Usually we use internal ones to prevent app full re-render and allow users to decide if they wanted to open it in new tab, but there is nothing wrong with leaving them as is, if we want to force opening in new tab.